### PR TITLE
do not show archive name in error msgs referring to repository

### PR DESCRIPTION
**Pull request by sw9719**
Fixes [issue](https://github.com/borgbackup/borg/issues/6014) where the error message would display the archive when the Respository does not exist by removing the archive name from the error string when `Repository.DoesNotExist`
is raised.

### CODE REVIEW

File: src/borg/helpers/parseformat.py [hunk 1]
```diff
@@ -492,6 +492,10 @@ def with_timestamp(self, timestamp):
             'utcnow': DatetimeWrapper(timestamp),
         })
 
+    def omit_archive(self):
+        #self.orig = self.orig.split("::")[0]
+        return self
```
Comment (ThomasWaldmann) [inline src/borg/helpers/parseformat.py:497]: This must be something like:
```
loc = Location(...)  # create new instance
... # modify instance as needed
return loc
```

**Comment by sw9719** [comment]
I did modify why the logic as mentioned. I hope this works.

File: src/borg/helpers/parseformat.py [hunk 2]
```diff
@@ -493,8 +493,9 @@ def with_timestamp(self, timestamp):
         })
 
     def omit_archive(self):
-        #self.orig = self.orig.split("::")[0]
-        return self
+        loc = Location(self.orig)
+        loc.orig = loc.orig.split("::")[0]
+        return loc
```
Comment (ThomasWaldmann) [inline src/borg/helpers/parseformat.py]: can you please add a unit test that checks whether this works as expected?

also check the value of `loc.archive` within that test.

Comment (sw9719) [reply src/borg/helpers/parseformat.py]: I am trying to figure out where to write the new test. Does testsuite/helpers.py sound good?

File: src/borg/testsuite/repository.py [hunk 3]
```diff
@@ -856,7 +856,7 @@ class RemoteRepositoryTestCase(RepositoryTestCase):
     repository = None  # type: RemoteRepository
 
     def open(self, create=False):
-        return RemoteRepository(Location('__testsuite__:' + os.path.join(self.tmppath, 'repository')),
+        return RemoteRepository(Location('__testsuite__:' + os.path.join(self.tmppath, 'repository')+"::archive"),
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/repository.py:859]: why is this needed?

Comment (sw9719) [reply src/borg/testsuite/repository.py:859]: Sorry, this was left over from something I was testing. Ill be more careful about this.

**Comment by sw9719** [comment]
I added tests for when repo does not contain archive name

File: src/borg/testsuite/helpers.py [hunk 4]
```diff
@@ -234,6 +234,12 @@ def test_bad_syntax(self):
             # this is invalid due to the 2nd colon, correct: 'ssh://user@host/path'
             Location('ssh://user@host:/path')
 
+    def test_omit_archive(self):
+        loc = Location('ssh://user@host:1234/some/path::archive')
+        loc_without_archive = loc.omit_archive()
+        assert loc_without_archive.archive == "archive"
```
[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply src/borg/testsuite/helpers.py:240]: did you see this ^^^?

[the remark this answers is not in the document]
Comment (sw9719) [reply src/borg/testsuite/helpers.py:240]: So I checked this. When '::<archive'-name> is present in the Location string, `Locatoin.archive` attribute will contain the string after '::".

With 'Location('ssh://user@host:1234/some/path')', i.e. without '::<archive-name', `Location.archive` attributed is `None.`

So I added tests for both test cases: When archive name is present in string and when it's not.

Since only `Location.orig` is being used to raise appropriate exceptions, I only modified that attribute. If needed I can change the value of `Location.archive` when returning from `omit_archive()` method, but I hope it does not break anything.

[the remark this answers is not in the document]
Comment (ThomasWaldmann) [reply src/borg/testsuite/helpers.py:240]: OK, so please set `.archive` to None within `omit_archive`.

[the remark this answers is not in the document]
Comment (sw9719) [reply src/borg/testsuite/helpers.py]: Done

**Comment by ThomasWaldmann** [comment]
Thanks! Can you check if the same change applies to 1.1-maint branch?

**Comment by sw9719** [comment]
> Thanks! Can you check if the same change applies to 1.1-maint branch?

I don't see the new test in 1.1-maint branch. Also, `src/borg/helpers/` dir which contained changes in `Location` class in `parseformat.py`.

**Comment by ThomasWaldmann** [comment]
`borg.helpers` used to be a module (`helpers.py`) in 1.1 and was split up into a package (directory) in master.

**Comment by hexagonrecursion** [comment]
I'll work on backporting this
